src/SignalR/clients/csharp/Client.Core/src/HubConnection.cs

```diff
@@ -116,6 +117,14 @@ public partial class HubConnection
         /// </summary>
         public TimeSpan HandshakeTimeout { get; set; } = DefaultHandshakeTimeout;
 
+        /// <summary>
+        /// Get the connections current connection Id.
```

Comment:
connection's

Reply:
`Get the connection's current Id.` reads better IMO

Reply:
Will ConnectionId be null in some cases or has that changed?

A null connection ID for an active connection will confuse some people. It should be mentioned in the comments.

Reply:
I agree we should document when this can be null. Is this only when the client is disconnected? Or is it also possible to establish a connection without negotiating using HubConnection?

Reply:
I believe a HubConnection with the transport explicitly specified as WebSockets will skip negotiation and never have the connection ID sent to the client.

Reply:
Yeah, when negotiation is skipped the client connection Id will be null as well as when the client is disconnected. I can file a docs issue so that we follow up once we get this change in across the clients.

Reply:
Whether or not we decide this is worth mentioning in a doc article, I think we should also mention this in the doc comments. Something similar to what you have for the Java client should be fine. E.g.

```
/// Gets the connections current Id. This value will be cleared when the connection is stopped and will have a new value every time the connection is (re)established.
/// This value will be null if the WebSockets transport is explicitly specified because the client skips negotiation in that case.
```

Reply:
I like that wording. We should also note that this will always be null when the `SkipNegotiation` flag is set to true.

Reply:
Good coverage w.r.t. docs. thanks!